src/model/NPMResolver.ts

```diff
@@ -0,0 +1,194 @@
+            this.clean();
+        }
+        this.hashFilePath = this.getHashFilePath();
+        if (!this.isNPMInstallVersion()) {
+            console.log(cgreen`⇢` + ` [NPM] package.json version is 1.0.0 -> no npm install needed / your node modules are checked in`);
+        } else {
+            console.log(cgreen`⇢` + ` [NPM] package.json version is 2.0.0 or higher -> checking if npm install is needed`);
+            this.checkAndInstall();
+            if (watch) {
+                this.registerWatchers();
+            }
+        }
+    }
+
+    private isNPMInstallVersion(): boolean {
+        return !(this.getPackageVersion() === '1.0.0');
+    }
+
+    private clean() {
+        try {
+            let path: string = this.getNodeModulesPath();
+            rimraf.sync(path);
+            console.log(cgreen`⇢` + ` [NPM] node_modules folder was cleaned/removed`);
+        } catch (e) {
+            console.error(cerr` ${e.message}`);
+            throw Error(e);
+        }
+    }
+
+    private registerWatchers() {
+        // packageLockWatcher
+        const packageJsonWatcher = chokidar.watch([this.getPackageLockPath()]);
+        packageJsonWatcher
+            .on('change', () => {
+                this.checkAndInstall();
+            })
+            .on('error', (e) => {
+                console.error(cerr`✗` + ` [packageLockWatcher] watcher failed: ${e}`);
+                nodeModulesWatcher.close();
+            });
+
+        // nodeModulesWatcher
+        const glob = Scheduler.convertToUnixPath(`${this.getNodeModulesPath()}/**/*`);
+        const nodeModulesWatcher = chokidar.watch(glob);
+        let ready: boolean = false;
+        let debounce: Timeout;
+
+        const handleEvent = () => {
+            if (!ready) {
+                return;
+            }
+            debounce && clearTimeout(debounce);
+            debounce = setTimeout(() => {
+                if (!fs.existsSync(this.getNodeModulesPath())) {
+                    console.log(cerr`✗` + ` [NPM] node_modules folder has been removed please restart Assets Compiler`);
+                    process.exit()
+                } else {
+                    this.checkAndInstall()
+                }
+            }, 500);
+        };
+        nodeModulesWatcher
+            .on('ready', () => ready = true)
+            .on('add', handleEvent)
+            .on('change', handleEvent)
+            .on('unlink', handleEvent)
+            .on('unlinkDir', handleEvent)
+            .on('error', (e) => {
+                console.error(cerr` [nodeModulesWatcher] watcher failed: ${e}`);
+                nodeModulesWatcher.close();
+            });
+    }
+
+    private checkAndInstall() {
+        if (this.hasNoNodeModules()) {
+            console.log(cgreen`⇢` + ` [NPM] There seems to be no installed Node Modules`);
+            this.doNpmInstall();
+        } else {
+            this.checkHashAndInstall();
+        }
+    }
+
+    private checkHashAndInstall() {
```

Comment:
I would prefer having the body of that method included in `checkAndInstall` since that method name is already perfect and the body doesn't grow too big.